### PR TITLE
PLAT-128502: Add UI tests for TooltipDecorator

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- added UI tests for agate/TooltipDecorator

### Links
[//]: # (Related issues, references)
PLAT-128502

### Comments

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]

**Review by daniel-stoian-lgp — APPROVED** [review]

File: tests/ui/apps/TooltipDecorator/TooltipDecorator-View.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+			Tooltip Button 1
+		</TooltipButton>
+		<TooltipButton
+			id="tooltipButton2"
+			backgroundOpacity="transparent"
+			selected
+			tooltipText="Hello Tooltip Button 2"
+			tooltipDelay={1000}
+		>
+			Tooltip Button 2
```
Comment (dongsuwon) [inline tests/ui/apps/TooltipDecorator/TooltipDecorator-View.js:30]: It is recommended that you clarify the purpose of using the button. e.g) Tooltip DelayedButton.

File: tests/ui/apps/TooltipDecorator/TooltipDecorator-View.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		</TooltipButton>
+		<TooltipButton
+			id="tooltipButton3"
+			disabled
+			size="small"
+			tooltipText="Hello Tooltip Button 3"
+		>
+			Tooltip Button 3
```
Comment (dongsuwon) [inline tests/ui/apps/TooltipDecorator/TooltipDecorator-View.js:38]: Same above

File: tests/ui/specs/TooltipDecorator/TooltipDecoratorPage.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	get isTooltipShowing () {
+		return element('.enact-fit.enact-clip.enact-untouchable>div', browser).isExisting();
+	}
+}
+
+class TooltipDecoratorPage extends Page {
+	constructor () {
+		super();
+		this.title = 'TooltipDecorator Test';
+		const tooltipDefault = new TooltipButtonInterface('tooltipButton1');
```
Comment (dongsuwon) [inline tests/ui/specs/TooltipDecorator/TooltipDecoratorPage.js:47]: tooltipDefaultButton or tooltipButtonDefault is better.
It gets confused with the tooltip.

File: tests/ui/specs/TooltipDecorator/TooltipDecoratorPage.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		const tooltipDelayed = new TooltipButtonInterface('tooltipButton2');
```
Comment (dongsuwon) [inline tests/ui/specs/TooltipDecorator/TooltipDecoratorPage.js:48]: Same above

File: tests/ui/specs/TooltipDecorator/TooltipDecorator-specs.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+const Page = require('./TooltipDecoratorPage');
+
+describe('TooltipDecorator', function () {
+
+	beforeEach(function () {
+		Page.open();
+	});
+
+	const {
+		tooltipDefault,
```
Comment (dongsuwon) [inline tests/ui/specs/TooltipDecorator/TooltipDecorator-specs.js:10]: tooltipDefaultButton or tooltipButtonDefault is better.
It is confused with the tooltip.

File: tests/ui/specs/TooltipDecorator/TooltipDecorator-specs.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		tooltipDelayed,
```
Comment (dongsuwon) [inline tests/ui/specs/TooltipDecorator/TooltipDecorator-specs.js:11]: Same above

**Comment by dongsuwon** [comment]
Jenkins passed:
enact-ui-tests/935/

**Pull request merged**